File: my_solutions/1628.design-an-expression-tree-with-evaluate-function/solution.java
/**
 * This is the interface for the expression tree Node.
 * You should not remove it, and you can define some classes to implement it.
 */

abstract class Node {
    public abstract int evaluate();
    // define your fields here
    protected String val;
    protected Node left;
    protected Node right;
};

class MyNode extends Node {
    public MyNode(String val) {
        this.val = val;
    }

    public int evaluate() {
        // TODO: implement
    }
}

File: my_solutions/1628.design-an-expression-tree-with-evaluate-function/solution.cpp
#include <vector>
#include <string>
#include <stack>
using namespace std;

class Node {
public:
    virtual ~Node(){};
    virtual int evaluate() const = 0;

protected:
    // define your fields here
    string val;
    Node* left;
    Node* right;
};

class MyNode : public Node {
public:
    MyNode(string val) {
        // TODO: implement
    }
};
